libtsuba/src/RDG.cpp

```diff
@@ -385,17 +385,25 @@ tsuba::RDG::DoMake(const katana::Uri& metadata_dir) {
       return part_result.error();
     }
 
-    if (local_to_global_id_ != nullptr) {
-      if (local_to_user_id_ == nullptr) {
-        // for backward compatibility
-        // NB: this is a zero-copy slice, so the underlying data is shared
-        set_local_to_user_id(local_to_global_id_->Slice(0));
-      } else if (local_to_user_id_->length() != local_to_global_id_->length()) {
-        KATANA_LOG_DEBUG(
-            "Number of User Node IDs {} do not match number of Global Node IDs "
-            "{}",
-            local_to_user_id_->length(), local_to_global_id_->length());
-        return tsuba::ErrorCode::InvalidArgument;
+    if (local_to_user_id_->length() == 0) {
+      // for backward compatibility
+      // NB: this is a zero-copy slice, so the underlying data is shared
+      set_local_to_user_id(local_to_global_id_->Slice(0));
+    } else if (local_to_user_id_->length() != local_to_global_id_->length()) {
+      if (local_to_global_id_ != nullptr) {
+        if (local_to_user_id_ == nullptr) {
```

Comment:
Both of these checks are superfluous. If either of them are `nullptr`, then `->length()` above should crash. In this case, the constructor guarantees that they are not `nullptr`.

Reply:
Sorry, I blindly threw that test in there and you are right, the logic should be cleaned up.  Done.